src/pugixml.cpp

```diff
@@ -6077,6 +6094,22 @@ namespace pugi
 		return true;
 	}
 
+	PUGI__FN bool xml_node::remove_children()
+	{
+		for (xml_node_struct* child = this->_root->first_child; child; child = child->next_sibling)
+		{
+			if (!_root || !child || child->parent != _root) return false;
+
+			impl::xml_allocator& alloc = impl::get_allocator(_root);
+			if (!alloc.reserve()) return false;
+
+			impl::destroy_node(child, alloc);
```

Comment:
This destroys the node, potentially deallocating storage and invalidating next_sibling. You have to read next_sibling *before* the loop. Please read the implementation of destroy_node, it has the loop that's exactly the same.